tools/integration-test/src/tests/ica_filter.rs

```diff
@@ -0,0 +1,258 @@
+use std::str::FromStr;
+
+use serde::Serialize;
+
+use ibc::core::ics04_channel::channel::State;
+use ibc_relayer::config::{
+    default::connection_delay as default_connection_delay,
+    filter::{ChannelFilters, FilterPattern},
+    PacketFilter,
+};
+
+use ibc_test_framework::{
+    bootstrap::binary::connection::bootstrap_connection,
+    ibc::denom::Denom,
+    prelude::*,
+    relayer::channel::{assert_eventually_channel_established, query_channel_end},
+};
+
+#[test]
+fn test_ica_filter_allow() -> Result<(), Error> {
+    run_binary_chain_test(&IcaFilterTestAllow)
+}
+
+pub struct IcaFilterTestAllow;
+
+impl TestOverrides for IcaFilterTestAllow {
+    // Use `icad` binary and deterministic identifiers for clients, connections, and channels
+    fn modify_test_config(&self, config: &mut TestConfig) {
+        config.bootstrap_with_random_ids = false;
+    }
+
+    // Enable channel workers and allow relaying on ICA channels
+    fn modify_relayer_config(&self, config: &mut Config) {
+        config.mode.channels.enabled = true;
+
+        for chain in &mut config.chains {
+            chain.packet_filter = PacketFilter::Allow(ChannelFilters::new(vec![(
```

Comment:
Instead of define another struct, I just added the `PacketFilter` as an argument to the existing test struct. Also, `config.mode.channels.enabled = true` is required and the default test config therefore won't work.